### DISCUSSION TITLE
Q&A + Ideas

### DISCUSSION THREAD

**Discussion by janpaulpl** [jpVinnie/RNAfoldml#1] [Q&A]
## 👋 Welcome!
  * Ask questions you’re wondering about.
  * Share ideas.

**Comment by janpaulpl** [comment]
@imbush Doubt regarding `fold.ml`:

What is happening in line 18 with a redundant `if else`? 

```Ocaml
(** [nussinov r] is the secondary structure for [r] given by Nussinov's
    algorithm to maximize pairing. *)
let nussinov (r : Rna.rna) = if r = r then temp else temp
```

**Reply by imbush** [reply]
These are temporary definitions to be replaced

**Comment by imbush** [comment]
# [FASTA Formatting Guide](https://en.wikipedia.org/wiki/FASTA_format#:~:text=In%20bioinformatics%20and%20biochemistry%2C%20the,comments%20to%20precede%20the%20sequences.)

**Reply by janpaulpl** [reply]
https://github.com/jpVinnie/RNAfoldml/commit/8babcacc3bfc9f27ea95e5c6623b0d2aebfb0461#commitcomment-68800201

**Comment by janpaulpl** [comment]
@imbush 

Is there any advantage on having these helper functions down here? Also why is `get_seq` and `get_name` the same? 
```Ocaml
let rep_ok r =
  match Str.search_forward (Str.regexp "\\([AGCU]+\\)") r.name 0 with
  | 0 -> r
  | _ -> failwith "RI"

...

let get_seq r = r.name
let get_info r = r.attributes
let get_name r = r.name
```

**Reply by janpaulpl** [reply]
Ok I got the fix! The functions are only for the `.mli`
```ocaml
let rep_ok r =
  match Str.search_forward (Str.regexp "\\([AGCU]+\\)") r.seq 0 with
  | 0 -> r
  | _ -> failwith "RI"

...

let get_seq r = r.seq
let get_info r = r.attributes
let get_name r = r.name
```
